src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -343,6 +357,48 @@ private void validateOperationProperties(@Nullable Schema schema, FailureCollect
           .withConfigElement(NAME_TABLE_KEY, keyField);
       }
     }
+
+    Map<String, Integer> keyMap = new HashMap<>();
+    keyFields.forEach(key -> {
+      Integer count = keyMap.get(key);
+      keyMap.put(key, count == null ? 1 : count + 1);
+    });
+
+    keyMap.keySet().stream()
+      .filter(key -> keyMap.get(key) != 1)
+      .forEach(key -> collector.addFailure(
+        String.format("Table key field '%s' is duplicated.", key),
+        String.format("Remove duplicates of Table key field '%s'.", key))
+        .withConfigElement(NAME_TABLE_KEY, key)
+      );
+
+    if (Operation.UPSERT.equals(getOperation()) && getOrderedBy() != null) {
+      List<String> orderedByFields = Arrays.stream(Objects.requireNonNull(getOrderedBy()).split(","))
+        .map(String::trim).collect(Collectors.toList());
+      orderedByFields.stream()
+        .filter(orderedByField -> !fields.contains(orderedByField.split(":")[0]))
+        .forEach(orderedByField -> collector.addFailure(
+        String.format("Ordered by field '%s' does not exist in the schema.", orderedByField.split(":")[0]),
+        "Change the Ordered by field to be one of the schema fields.")
+        .withConfigElement(NAME_ORDERED_BY, orderedByField));
+
+      Map<String, Integer> orderedByFieldMap = new HashMap<>();
+      Map<String, String> orderedByFieldValueMap = new HashMap<>();
+      orderedByFields.forEach(keyValue -> {
+          String key = keyValue.split(":")[0];
+          Integer count = orderedByFieldMap.get(key);
+          orderedByFieldMap.put(key, count == null ? 1 : count + 1);
```

Comment:
you don't need to count to check that field already exists in a map. Or you can extract a method to calculate number of duplicates and reuse it in both cases.

Reply:
Done.